### ISSUE TITLE
单卡下运行pretrain.py 报错 Default process group has not been initialized, please make sure to call init_process_group.

### ISSUE THREAD

**Issue by TristanShao** [DLLXW/baby-llama2-chinese#18]
是因为单卡运行了多卡模式吗。但是不知道怎么改。

**Comment by TristanShao** [comment]
奇怪，把ddp输出也是false

**Comment by yiyiyichen1** [comment]
pretrain里加上下面代码
import torch.distributed as dist
dist.init_process_group('gloo', init_method='file:///tmp/somefile', rank=0, world_size=1)

**Comment by vaderyang** [comment]
删除get_rank那段出错的if，直接存就可以

**Comment by Niculuse** [comment]
单卡训练，注释下面这行
`train_sampler = torch.utils.data.distributed.DistributedSampler(train_ds)`

然后把这里的sampler设置为None

    train_loader = torch.utils.data.DataLoader(
        train_ds,
        batch_size=batch_size,
        pin_memory=False,
        drop_last=False,
        shuffle=False,        
        num_workers=4,
        sampler=None
    )

**Comment by Vincent-ZHQ** [comment]
单卡运行也按照多卡运行的命令写，指定参数为1就好了，不要直接python pretrain.py

**Comment by van-68** [comment]
> 单卡运行也按照多卡运行的命令写，指定参数为1就好了，不要直接python pretrain.py

你好，请问，这个具体是怎么执行和设置呢，谢谢。

**Comment by zerozhoujie** [comment]
加载数据下面加上这几行

if ddp:
        train_sampler = torch.utils.data.distributed.DistributedSampler(train_ds)
else:
        train_sampler = None

**Comment by Tonikroosliruhao** [comment]
> > 单卡运行也按照多卡运行的命令写，指定参数为1就好了，不要直接python pretrain.py
> 
> 你好，请问，这个具体是怎么执行和设置呢，谢谢。

`torchrun` --standalone --nproc_per_node=1 pretrain.py`

**Comment by Zha-Miku** [comment]
> 删除get_rank那段出错的if，直接存就可以

实测运行成功
![image](https://github.com/DLLXW/baby-llama2-chinese/assets/54501325/6229e127-ca96-4aaa-8d5a-6a06813e5d08)


具体操作：直接注释掉`if torch.distributed.get_rank() == 0:`这行就行了

系统：win11，py3.10，3050m，cu118

就是有点不理解具体为啥原因，

代码本来应该是在Linux上面运行的，而且命令只能在Linux上可行，但是win上就不行了，不理解

**Comment by HildaM** [comment]
按照大佬们的修改，注释掉多卡训练的内容，还是会出现以下的错误：
![38e2c2c015ff0943f321e20f45529ae7](https://github.com/DLLXW/baby-llama2-chinese/assets/46062705/18e97bf8-8458-4cea-b6ad-7c4af8a6ebc1)

**Comment by 18207809132** [comment]
这报错的解决方法：将代码行train_sampler = torch.utils.data.distributed.DistributedSampler(train_ds)改为train_sampler = None即可。
The solution to this error: Change the code line” train_sampler = torch.utils.data.distributed.DistributedSampler(train_ds) “to ”train_sampler = None“.
